lib/cocoapods/generator/prepare_artifacts_script.rb

```diff
@@ -0,0 +1,248 @@
+require 'cocoapods/xcode'
+
+module Pod
+  module Generator
+    class PrepareArtifactsScript
+      # @return [Hash{String => Array<Pod::Xcode::XCFramework>}] Multiple lists of xcframeworks per
+      #         configuration.
+      #
+      attr_reader :xcframeworks_by_config
+
+      # @return [Pathname] the root directory of the sandbox
+      #
+      attr_reader :sandbox_root
+
+      # @return [Platform] the platform of the target for which this script will run
+      #
+      attr_reader :platform
+
+      # @param  [Hash{String => Array<Pod::Xcode::XCFramework>] xcframeworks_by_config
+      #         @see #xcframeworks_by_config
+      #
+      # @param  [Pathname] sandbox_root
+      #         the sandbox root of the installation
+      #
+      # @param  [Platform] platform
+      #         the platform of the target for which this script will run
+      #
+      def initialize(xcframeworks_by_config, sandbox_root, platform)
+        @xcframeworks_by_config = xcframeworks_by_config
```

Comment:
the class name is `PrepareArtifactsScript` but it is mostly dealing with XCFrameworks, should we just rename this to `PrepareXCFrameworksScript`?

Reply:
I initially named it something like that but renamed it since this could be used to copy any artifact that needs to be in the build folder before the `Compile Sources` build phase. not strongly opposed to changing it but feels like generic is better in this case